react/getting_started_with_react/keys_in_react.md

```diff
@@ -0,0 +1,286 @@
+### Introduction
+
+In this lesson we will cover what keys are in React, rendering a list, and touch on when and how React knows to render or re render.
+
+### Lesson Overview
+
+by the end of the lesson you should be able to answer the following:
+
+- Why does React need keys
+
+- How to use keys
+
+- How do keys impact rendering, and re-rendering.
+
+- Where to get keys
+
+- Patterns to avoid when using keys
+
+- Can we use index as a key
+
+
+### Why does React need keys?
+
+Lets imagine you take notes while working through The Odin Project.
+
+Maybe you use index cards, or lined paper.
+
+Now if these notes are in a pile, we can think of them as an `array` and each one having a particular `index` in the list.
+
+The paper at the bottom of the pile is the first note you made or `note[0]`, the one directly on top of it being the second note or `note[1]`.
+
+What happens if we want to reorder the list, or remove a note from the list. Maybe we need to find an old note in the middle of the pile.
+
+If `note[4]` was removed, does that mean `note[5]` just became `note[4]`? does every note after `note[5]` have to be re indexed?
+
+If we reorder or *sort* the notes, we might end up in a situation where our previous `note[14]` might become our new `note[3]`.
+
+It gets a bit confusing to manage and keep track of which papers are still in order.
+
+One solution might be to add a unique summary or `key` of some kind to each page.
+
+This way no matter where that specific `note` moves to in the pile, we know which note it is.
+
+Similarly `keys` help React to keep track of the items in a list. `keys` allow React to know when to render, what to render, when to re render, and when not to re render.
```

Comment:
Like the example here, but we normally discourage taking notes while working through TOP. It would be contradictory to add this example. We could slightly modify it perhaps